### ISSUE TITLE
False positive IDE0047: Parentheses can be removed

### ISSUE THREAD

**Issue by a510** [dotnet/roslyn#50025]
(Based on **Alexander Shvedov**'s [tweet](https://twitter.com/controlflow/status/1339253767221485568?s=20).)

**Version Used**: 
VS 16.8.3

**Steps to Reproduce**:
``` csharp
public class C
{    
    public const int Foo = 1;  
    
    public void M(object o)
    {
        if (o is (Foo)) M(1);  // IDE0047 
    }    
}

public class Foo { }
```

**Expected Behavior**:
IDE0047 fix should not be suggested because once the parentheses are removed the `is` operator will match against the `Foo` type instead of the `Foo` constant.

**Actual Behavior**:
IDE0047 fix is suggested.

**Comment by vorotynsky** [comment]
Should a constant check be always in parentheses?

For example, we have this code
```cs
using MegaLib;

public class C
{    
    public const int Foo = 1;  
    
    public void M(object o)
    {
        if (o is Foo) M(1);  // constant check
    }    
}
```
MegaLib v1.0 has no `Foo` class, but MegaLib v1.1 has `Foo`. An update *brokes* this code.

**Comment by CyrusNajmabadi** [comment]
No.  There are many cases where updates to libs would break downstream code.  Analyzers would not be viable if they had to be resilient to literally any future code.
